test/format/ops.c

```diff
@@ -535,60 +533,32 @@ begin_transaction(TINFO *tinfo, WT_SESSION *session, u_int *iso_configp)
 	testutil_check(ret);
 
 	if (v == ISOLATION_SNAPSHOT && g.c_txn_timestamps) {
-		/* Avoid starting a new reader when a prepare is in progress. */
+		/*
+		 * Prepare will return an error if the prepare timestamp is less
+		 * than any active read timestamp. Lock across allocating read
+		 * timestamps if prepare is configured, skip the read timestamp
+		 * if a prepare is in progress,
+		 */
+		locked = false;
 		if (g.c_prepare) {
-			testutil_check(pthread_rwlock_rdlock(&g.prepare_lock));
+			ret = pthread_rwlock_tryrdlock(&g.prepare_lock);
```

Comment:
What's wrong with waiting here?  Is there some operation that can deadlock, or that holds a write lock indefinitely?

Reply:
The reason not to wait here is waiting serializes transaction-begin with transaction-prepare, and it seemed potentially interesting to mix-and-match reads with and without timestamps. I don't have any data to show it's worth doing.

Just let me know if you prefer to always wait.

Reply:
I'd prefer to wait here, then schedule a ticket to allow `WT_SESSION::timestamp_transaction` to set the prepare timestamp, and just do that part under the lock.